docs/development/pigs/pig-007.rst

```diff
@@ -0,0 +1,409 @@
+morphology or spatial models, combined spectral and spatial models (`SkyModel`)
+and lightcurve or phase models. We propose a uniform naming scheme for these model
+classes, based on the prefixes `Spectral...`, `Spatial...`, `Temporal`:
+
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|   Base class    | Template class                               | IRF folded class     | Model classes               |
++=================+==============================================+======================+=============================+
+|SpectralModel    |SpectralTemplate                              |SpectralIRFModel      | Powerlaw, LogParabola, ...  |
+|(SpectralModel)  |(TableModel)                                  |                      |                             |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|SpatialModel     |SpatialTemplate                               |SpatialIRFModel?      | Shell, SpatialGaussian, ... |
+|(SkySpatialModel)|(SkyDiffuseMap)                               |                      |                             |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|TemporalModel    |PhaseCurveTemplate, LightCurveTemplate        |                      | Sine, ...                   |
+|(None)           |(PhaseCurveTableModel), (LightCurveTableModel)|                      |                             |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|SourceModel      |SourceTemplate                                |SourceIRFModel        |Arithmetic spectral/spatial  |
+|(SkyModel)       |(SkyDiffuseCube)                              |                      |models?                      |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+|BackgroundModel  |BackgroundTemplate?                           |BackgroundIRFModel    |Arithmetic background        |
+|(None)           |(None)                                        |      (None)          |models?                      |
++-----------------+----------------------------------------------+----------------------+-----------------------------+
+
+The current class names are given in parentheses. Optional model classes, where
+the need is unclear are marked with a question mark.
+
+Parametric models are named after their parametrization, such as `Powerlaw` or
+`Gaussian`. In case the name is not unique (e.g. for a gaussian spectral and spatial
+model, the corresponding prefix should be used, such as `SpectralGaussian` or
+`SpatialGaussian` or `SpatialConstant` and `SpectralConstant`).
+
+In addition there is the `SourceModels`  class to represent a sum of `SourceModel`
+objects.
```

Comment:
I think for v1.0 it's probably not needed to support multiple (hadronic) background models. Technically astrophysical background models (IEM) are already supported via the `SourceTemplate` class. Long-term I agree it might make sense to introduce support for multiple background models. I'll re-think this part to see if there is a need for such an abstraction now, if not I'll add a sentence to explain this.